src/main/java/io/pinecone/clients/Inference.java

```diff
@@ -0,0 +1,40 @@
+package io.pinecone.clients;
+
+import org.openapitools.control.client.ApiException;
+import org.openapitools.control.client.api.InferenceApi;
+import org.openapitools.control.client.model.EmbedRequest;
+import org.openapitools.control.client.model.EmbedRequestInputsInner;
+import org.openapitools.control.client.model.EmbedRequestParameters;
+import org.openapitools.control.client.model.EmbeddingsList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Inference {
+    InferenceApi inferenceApi;
+
+    public Inference() {
+        inferenceApi = new InferenceApi();
+    }
+
+    public EmbeddingsList embed(String model, String truncate, String inputType, List<String> inputs) throws ApiException {
+        EmbedRequestParameters embedRequestParameters = new EmbedRequestParameters().inputType(inputType);
+        if(truncate != null && !truncate.isEmpty())
+            embedRequestParameters.truncate(truncate);
+        List<EmbedRequestInputsInner> EmbedRequestInputsInnerList = convertInputStringToEmbedRequestInputsInner(inputs);
+        EmbedRequest embedRequest = new EmbedRequest()
+                .model(model)
+                .parameters(embedRequestParameters)
+                .inputs(EmbedRequestInputsInnerList);
+
+        return inferenceApi.embed(embedRequest);
+    }
+
+    private List<EmbedRequestInputsInner> convertInputStringToEmbedRequestInputsInner(List<String> inputs) {
+        List<EmbedRequestInputsInner> embedRequestInputsInnerList = new ArrayList<EmbedRequestInputsInner>();
+        for(String input:inputs) {
+            embedRequestInputsInnerList.add(new EmbedRequestInputsInner().text(input));
+        }
+        return embedRequestInputsInnerList;
+    }
```

Comment:
This would be better using stream API: 

```java
inputs.stream()
                 .map(input -> new EmbedRequestInputsInner().text(input))
                 .collect(Collectors.toList());
```

Reply:
Why do you think using stream is better here? Wouldn't it create a new object for intermediate step?

Reply:
I'd have to look more closely at the underlying impl, but I think using the stream impl gives java flexibility to optimize how/when the underlying list actually gets created, and it may never have to allocate a new list at all depending on the usage.

Reply:
This is your call, and don't think it'll make much difference in how fast things run on such small datasets. I just find the stream api more pleasant to use, and gives java an opportunity to optimize what's happening under the hood more so than if you explicitly instantiate a list. But this is not a showstopper by any means